_posts/2016-10-31-angular2-discussion-portal.md

```diff
@@ -0,0 +1,141 @@
+date:   2016-10-31 08:00:00 +0200
+author: "Tomasz Banaś"
+tags: angular angular2 gulp tslint rxjs
+---
+
+Angular2 was released October this year. On the Internet you can find more and more tutorials which shows how to build a simple application. This tutorial is quite different from them. It shows how to build an Angular 2 application where we're going to focus on architecture which will periodically pull data from backend to keep fresh data on UI.
+
+__After this tutorial you will know:__
+
+* how to set up angular2 project (using gulp + tslint) and build discussion portal,
+* how to build a proper architecture (using angular2 + rxjs),
+* how to create components which will periodically pull date from backend,
+
+Let's start with watching a quick demo how this project works.
+<img src="/images/blog/posts/angular2-discussion-portal/demo.gif"/>
+
+# 1 Set up project
+
+The whole project is available on [github](https://github.com/BeyondScheme/angular2-discussion-portal). Please follow instructions written in a [README](https://github.com/BeyondScheme/angular2-discussion-portal/blob/master/README.md) file to run the project. 
+
+What we are going to use while building project:
+
+* lite-server - server which we will use to run the project,
+* json-server - it is needed for simple db with restful api (we'll share data between each app instance so that's why I didn't choose `angular-in-memory-web-api`),
+* gulp - tool to build our application,
+* tslint - checkstyle for typescript. I strongly recommend to use this tool - it will fail a build when some checkstyle rules will be violated.
+
+I'm going to briefly discuss each config file:
+ 
+1. `bs-config.json` Configuration file for lite-server. `ghost-mode` flag disabled synchronization of browsers and files. We're going to test our application on two instances so synchronization should be disabled.
+2. `db.json` Our database. This file is used by json-server.
+3. `gulfile.js` Config file for gulp. We have three tasks:
+    * `watch` - monitoring files changes and rebuilding if necessary,
+    * `tslint, build-ts, build-css` - tasks for running tslint, building typescripts and css files,
+4. `package.json` Contains npm dependencies for project.
+5. `systemjs.config.js` Provides information to a module loader about where to find application modules, and registers all the necessary packages.
+6. `tsconfig.json`Defines how the TypeScript compiler generates JavaScript from the project's files.
+7. `tslint.json` Contains checkstyle rules for tslint.
+8. `typings\globals\es6-shim\index.d.ts` Typings needed by typescript.
+
+# 2 Architecture
+
+<img src="/images/blog/posts/angular2-discussion-portal/architecture.png"/>
+
+As you can see on image we have two components:
+
+#### __DashboardComponent__
+This component is responsible for displaying existing posts and an input for creating new one. Data for this component are provided by `PostService`.
```

Comment:
You can add link to files that are not in the blog post content so it will be easy to just click and see what's the content of particular component or the service.

I had to browse repo to find where is https://github.com/BeyondScheme/angular2-discussion-portal/blob/master/src/ts/shared/services/post.service.ts